In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("the-reddit-climate-change-dataset-posts.csv")

In [3]:
df

type      id subreddit.id       subreddit.name  subreddit.nsfw  \
0       post  x2smmw        2vh0v  democraticsocialism           False   
1       post  x2slxy        2w844    nostupidquestions           False   
2       post  x2slk5       6wzx9b           eduwriters           False   
3       post  x2sbz6       46adh7  alternatitvereality           False   
4       post  x2sa2a        2qh1i            askreddit           False   
...      ...     ...          ...                  ...             ...   
620903  post   aki85        2qh1n          environment           False   
620904  post   akgtp        2qk5q      climateskeptics           False   
620905  post   akgkb        2qk5q      climateskeptics           False   
620906  post   akgbz        2qk5q      climateskeptics           False   
620907  post   ake6l        2qh1i            askreddit           False   

        created_utc                                          permalink  \
0        1661990237  https://old.reddit.com/r/DemocraticSocialism/c...   
1        1661990182  https://old.reddit.com/r/NoStupidQuestions/com...   
2        1661990150  https://old.reddit.com/r/EduWriters/comments/x...   
3        1661989389  https://old.reddit.com/r/AlternatitveReality/c...   
4        1661989240  https://old.reddit.com/r/AskReddit/comments/x2...   
...             ...                                                ...   
620903   1262350233  https://old.reddit.com/r/environment/comments/...   
620904   1262333495  https://old.reddit.com/r/climateskeptics/comme...   
620905   1262330777  https://old.reddit.com/r/climateskeptics/comme...   
620906   1262328753  https://old.reddit.com/r/climateskeptics/comme...   
620907   1262304750  https://old.reddit.com/r/AskReddit/comments/ak...   

                         domain  \
0                   twitter.com   
1        self.nostupidquestions   
2                eduwriters.pro   
3                   i.imgur.com   
4                self.askreddit   
...                         ...   
620903  seminal.firedoglake.com   
620904             examiner.com   
620905             ritholtz.com   
620906      wattsupwiththat.com   
620907                 kusi.com   

                                                      url  \
0       https://twitter.com/SusanofTexas/status/156480...   
1                                                     NaN   
2       https://eduwriters.pro/?cid=2860/?utm_source=r...   
3                         https://i.imgur.com/wJbchkQ.jpg   
4                                                     NaN   
...                                                   ...   
620903         http://seminal.firedoglake.com/diary/21707   
620904  http://www.examiner.com/x-25061-Climate-Change...   
620905  http://www.ritholtz.com/blog/2009/12/statistic...   
620906  http://wattsupwiththat.com/2009/12/31/ultimate...   
620907  http://www.kusi.com/home/78477082.html?video=p...   

                                                 selftext  \
0                                                     NaN   
1       Ok so I was having a conversation with my neig...   
2                                                     NaN   
3                                                     NaN   
4                                               [deleted]   
...                                                   ...   
620903                                                NaN   
620904                                                NaN   
620905                                                NaN   
620906                                                NaN   
620907                                                NaN   

                                                    title  score  
0       “Cities without water” is our future. The lead...     15  
1                               Sharks and climate change      2  
2                     How to start a climate change essay      1  
3       Ever notice that when it’s hot a few days in J...      1  
4       

In [7]:
df.shape

(620908, 12)

In [9]:
df.info

<bound method DataFrame.info of         type      id subreddit.id       subreddit.name  subreddit.nsfw  \
0       post  x2smmw        2vh0v  democraticsocialism           False   
1       post  x2slxy        2w844    nostupidquestions           False   
2       post  x2slk5       6wzx9b           eduwriters           False   
3       post  x2sbz6       46adh7  alternatitvereality           False   
4       post  x2sa2a        2qh1i            askreddit           False   
...      ...     ...          ...                  ...             ...   
620903  post   aki85        2qh1n          environment           False   
620904  post   akgtp        2qk5q      climateskeptics           False   
620905  post   akgkb        2qk5q      climateskeptics           False   
620906  post   akgbz        2qk5q      climateskeptics           False   
620907  post   ake6l        2qh1i            askreddit           False   

        created_utc                                          permalink  \
0    

In [17]:
posts = df[["title", "url", "score"]]
posts.sort_values(by=["score"], ascending=False).head(50)

title  \
145834     At A School Strike Protest For Climate Change.   
120968  When the maldivian president held the world's ...   
367402  19 of 20 World Leaders Just Pledged to Fight C...   
467760  A 100 year old paper article about 'climate ch...   
9361                Climate Change Denier Gets Demolished   
111644  White House Website Recognizes Climate Change ...   
251608  “Please save your praise. We don’t want it,” S...   
453     An attempt to embarrass a climate change activ...   
434919      Pope Francis denounces climate change deniers   
301722  Prince Charles spent 75 minutes longer than sc...   
450252  'We're trying to go all in': Chocolate giant M...   
138725     At A School Strike Protest For Climate Change.   
291342  On the Hottest Day in the History of France, P...   
154536  Bill Gates Thinks That The 1% Should Foot The ...   
70891                 Climate change prediction from 1912   
157500  ‘Human beings have overrun the world’: David A...   
279310  61% of Canadians want government to take actio...   
170633  Jeff Bezos: World's richest man pledges $10bn ...   
380367  Teen Climate Activist to Crowd of Thousands: '...   
65401   Climate change is not a political issue, it's ...   
75225   14,000 scientists warn of "untold suffering" i...   
336531  Resurfaced Video Shows Bernie Sanders Criticiz...   
115729  Global rich must cut their carbon footprint 97...   
451378  Pope Francis, who has a strong belief in the s...   
143249  Bill Gates issued a stark warning for the worl...   
273497    107 Years ago the Press reported Climate Change   
291513  Trump dismisses need for climate change action...   
162407  'The rich are to blame for climate change' int...   
431794  Donald Trump Not Invited to French Climate Cha...   
499684  The White House website's page on climate chan...   
110440  The '1%' are the main drivers of climate chang...   
308764  Scientists have found "supercorals" in Hawaii’...   
499801  All References to Climate Change Have Been Del...   
412895  'Money talks': A $1.2 trillion fund manager is...   
362918  Climate Change Conference Erupts Into Laughter...   
273486  "Climate change contrarians" are getting 49 pe...   
288049  David Attenborough has highlighted Australia a...   
472757  China tells Donald Trump there is an 'internat...   
98449   Progressive Leader to Biden: Don't 'Water Down...   
264863  Meteorologist has had enough of climate change...   
475070  The Pope gave Donald Trump a 192-page letter h...   
155824  Pakistan begins colossal tree planting campaig...   
195637  Badass Boomer responds to being Ok'ed by a jou...   
78021   Greenland suspends oil exploration because of ...   
234083  Climate change: Greta Thunberg calls out the '...   
333858  Memes should be part of the curriculum on clim...   
99085   Japan’s Kyoto cherry blossoms peak on earliest...   
216300  Elon Musk has donated $1 million worth of tree...   
298945  Radiohead Donates Money to Climate Change Grou...   
95079   Climate change is driving some to skip having ...   

                                                      url   score  
145834                https://i.redd.it/88m4bzcxnfb51.png  151210  
120968                https://i.redd.it/v92pf2tebgz51.jpg  150816  
367402  https://www.motherjones.com/environment/2018/1...  134983  
467760                 https://i.redd.it/zsi5uxhuxa2z.jpg  133300  
9361                  https://i.redd.it/uwo18o8wsqc91.jpg  130766  
111644                                                NaN  128747  
251608  https://www.thehindu.com/news/international/do...  117496  
453                       https://v.redd.it/gatfgxxlrqk91  104965  
434919  https://www.cbsnews.com/news/pope-francis-deno...   98784  
301722  https://amp.theguardian.com/us-news/2019/jun/0...   94859  
450252                                                NaN   89575  
138725                https://i.redd.it/1ywae4ajhnk51.jpg   89025  
291342  https://www.commondreams.org/news/2019/06/28/h..

Il post della riga 467760, quarto per punteggio tra i post su Reddit, rimanda a un articolo su rivista scientifica del 1912, tuttavia non è l'articolo più antico riguardo all'argomento. Secondo ChatGPT quello più antico risale al 1896 ed è stato pubblicato dal chimico svedese Svante Arrhenius. L'articolo, intitolato "On the Influence of Carbonic Acid in the Air upon the Temperature of the Ground" (Sull'influenza dell'acido carbonico nell'aria sulla temperatura del suolo), è stato pubblicato sulla rivista scientifica Philosophical Magazine and Journal of Science e descrive il processo di assorbimento della radiazione infrarossa da parte dell'anidride carbonica nell'atmosfera, e come questo potrebbe portare a un aumento della temperatura globale. Questo articolo è considerato **uno dei primi** studi scientifici sulla teoria del riscaldamento globale (reperibile al seguente indirizzo: https://www.rsc.org/images/Arrhenius1896_tcm18-173546.pdf).